### ISSUE TITLE
뉴럴넷의 가장 큰 단점은 무엇인가? 이를 위해 나온 One-Shot Learning은 무엇인가?

### ISSUE THREAD

**Issue by jarammm** [AInsteins/AI-Tech-Interview#59]

**Comment by kimyuuuuuu** [comment]
- 뉴럴넷을 이용하면 **고차원의 데이터에 대해 좋은 퍼포먼스**를 보인다.하지만, 일반적으로는 **학습데이터가 클 때만 해당**되는 말.
- **One-shot Learning** 은 뉴럴넷도 새로운 레이블을 지닌 데이터가 적을 때 (one-shot 에서는 한 개) 에도 모델이 좋은 성능을 내도록 사용되는 방법이다.
- 이를 위해서는 기존에 다른 레이블의 많은 데이터를 학습하여 데이터의 특성을 잘 이해하는 **pretrained** 모델이 필요하다.
- 학습된 모델에 새로운 레이블의 데이터 하나 던져 주면 모델은 데이터의 특성에 대한 이해를 바탕으로 이 레이블에 대해서도 이해를 하게 된다.

**Comment by miinkang** [comment]
- 뉴럴넷의 단점 
    - 학습을 위한 많은 데이터가 필요하다.
- One-shot learning 
    - 데이터가 적을 때 모델이 좋은 성능을 내도록 사용되는 방법 
    - Pre-trained  모델이 필요함

**Comment by jarammm** [comment]
- 뉴럴넷은 처음 본 것이라도 한 번 보면 여타 종류들과 구분하는 것이 어렵지 않은 사람과 달리 대량의 데이터 학습을 요한다.
    - 모든 N개의 class에 대하여 학습을 진행할 만큼의 충분한 데이터가 필요하고
    - N개의 class를 분류할 수 있는 모델에서 기존에 없던 class의 데이터가 제시된 경우 모델을 처음부터 학습시켜야 함

- One-shot learning은 각 레이블당 K개의 데이터를 학습하는 방식인 Few shot learning의 극단적인 예시로 K가 1, 즉 label 당 이미지를 하나씩만 보여주고 학습하는 방법
- 예를 들어 하나의 사자 이미지를 보고, 다른 사자 이미지를 보여주었을 때 사자라고 분류할 수 있도록 학습하는 방법 → 사자의 특징을 잡아내고, 다른 동물들과 구분하는 법을 배우는 데에 초점을 둠

**Comment by import-max** [comment]
- 사람과 다르게, 처음 보는 것(label)을 구분하는 것이 어려우며 이를 위해서는 많은 사전학습이 필요하다.
- `One-shot Learning` 은 뉴럴넷도 새로운 label을 구분할 때 좋은 성능을 내도록 사용되는 방법으로, pretrained 모델이 필요하다.
